doc/HORDES.md

```diff
@@ -0,0 +1,62 @@
+existing hordes live in overmap::zg;
+hordes move with calls to overmap::move_hordes()
+These should both be removed, specific monster instances should be created instead of hordes, probably.
+
+monsters outside the reality bubble live in overmap::monster_map
+methods for spawning and despawning monsters live in overmapbuffer::{spawn,despawn}_monster()
+need an act method, overmapbuffer::move_hordes()
+In overmapbuffer since they will need to cross overmap borders.
+Store references to monsters on a queue keyed by turns, after acting monsters insert themselves in the queue a number of turns in the future based on their speed.
+Treat each submap of monsters as a group that gets processed at once, that would be slightly less turn-accurate, but have MUCH better memory coherence.
+
+Use cases:
+Steady-state processing:
+The overmap is triggered to do a turn, it grabs the monster group at the head of monster_turn_sequence and checks if they are due for processing, if they are, it processes them, moving to new locations in monster_map if necessary, and re-inserts them in the appropriate monster_turn_sequence (might be on a new overmap).
```

Comment:
Yes, we can guarantee that we process the "most out of date" monsters in each group.